### PR TITLE
Implement flatfile functionality for CNV and introduce cnvKompare

### PR DESCRIPTION
**Pull request by Kdreval**
This PR introduces the flatfile functionality for CNV to avoid reliance on the database. The useful feature here is a new `cnvKompare` function that will compare CNV data for patients with multiple time points and calculate the overall concordance of CNV + other comparison data. It can also handle the comparison of CNV between different tools if needed.

### CODE REVIEW

**Comment by Kdreval** [comment]
This also addresses and resolves issue #19

File: R/utilities.R [hunk 1]
```diff
@@ -2910,3 +2910,310 @@ adjust_ploidy = function(this_seg,
 
   return(this_seg)
 }
+
+
+#' This function will compare CNV data between samples with multiple time points. It can also handle same-sample comparison
```
Comment (lkhilton) [inline R/utilities.R:2915]: This format will put this entire block of text into the title so when you look at the help for this function all these details are in large bold font at the top. The format should go like this: 

```
#' Sum of vector elements
#'
#' `sum` returns the sum of all the values present in its arguments.
#'
#' This is a generic function: methods can be defined for it directly
#' or via the [Summary()] group generic. For this to work properly,
#' the arguments `...` should be unnamed, and dispatch is on the
#' first argument.
```

(From https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html)

Perhaps while you fix this you could update the docstrings for `sanitize_maf_data`?

**Review by lkhilton — CHANGES_REQUESTED** [review]
Awesome! Just needs an update to the docstrings.

File: R/utilities.R [hunk 2]
```diff
@@ -2910,3 +2910,310 @@ adjust_ploidy = function(this_seg,
...
+
+  # check that sample identifiers are provided
+  if (missing(patient_id) & missing(sample_ids)) {
+    stop("Please provide patient id or sample ids for comparison.")
+  }
+
+  # retrieve sample ids if only patient id is specified
+  if (missing(sample_ids)) {
+    sample_ids = get_gambl_metadata()
+    sample_ids = dplyr::filter(sample_ids, patient_id == {
```
[the remark this answers is not in the document]
Comment (Kdreval) [reply R/utilities.R:2965]: Ah, this is syntax autoformatting! Have to be careful not to miss this the next time!

File: R/utilities.R (hunk 1 above)
Comment (Kdreval) [reply R/utilities.R:2915]: Thanks, I'll fix this!

**Comment by Kdreval** [comment]
Thanks for review Laura! I fixed these issues, and also added additional function to specify minimum similarity between cytobands to be considered concordant/discordant

**Review by lkhilton — APPROVED** [review]
Looks great! Adding percent concordance as a configurable parameter is a great addition.

**Pull request merged**